R/safe2x2Test.R

```diff
@@ -1556,54 +1671,71 @@ simulateWorstCaseQuantileTwoProportions <- function(na, nb, priorValues,
-    #distribution.
-    #also store the implied target belonging to the worst case power: exp(Expected [log E-waarde])
     if (currentQuantile >= currentWorstCaseQuantile) {
       currentWorstCaseQuantile <- currentQuantile
-      if (expectedStopTime) {
-        bootResult <- boot::boot(
-          data = stoppingTimes,
-          statistic = function(x, idx) {
-            mean(x[idx])
-          },
-          R = bootN
-        )
-      } else {
-        bootResult <- boot::boot(
-          data = stoppingTimes,
-          statistic = function(x, idx) {
-            quantile(x[idx], 1 - beta)
-          },
-          R = bootN
-        )
-      }
-      worstCaseQuantileTwoSe <- 2*sd(bootResult[["t"]])
+      #store obtained stopping times for bootstrapping later
+      stoppingTimesWorstCase <- stoppingTimes
+
     }
     if (currentPower <= currentWorstCasePower) {
       currentWorstCasePower <- currentPower
-      bootResultPower <- boot::boot(
-        data = stopEs,
-        statistic = function(x, idx) {
-          mean(x[idx] >= 1/alpha)
-        },
-        R = bootN
-      )
-      powerTwoSe <- 2*sd(bootResultPower[["t"]])
-      currentImpliedTarget <- exp(mean(log(stopEs)))
+      stopEsWorstCase <- stopEs
+      finalEsWorstCase <- finalEs
     }
   }
   close(pbSafe)
 
-  return(list(worstCasePower = currentWorstCasePower,
-              worstCasePowerTwoSe = powerTwoSe,
-              worstCaseQuantile = currentWorstCaseQuantile,
-              worstCaseQuantileTwoSe = worstCaseQuantileTwoSe,
-              impliedTarget = currentImpliedTarget))
+  #bootstrapping to estimate standard deviation of metrics
+  if (expectedStopTime) {
+    bootResultNPlan <- computeBootObj(
+      values = stoppingTimesWorstCase,
+      nBoot = nBoot,
+      objType = "expectedStopTime"
+    )
+    worstCaseQuantileTwoSe <- 2 * bootResultNPlan[["bootSe"]]
+  } else if (beta != 0){
+    #beta is set to 0 if NPlan is already given, then do not estimate SE
+    bootResultNPlan <- computeBootObj(
+      values = stoppingTimesWorstCase,
+      beta = beta,
+      nBoot = nBoot,
+      objType = "nPlan"
+    )
+    worstCaseQuantileTwoSe <- 2 * bootResultNPlan[["bootSe"]]
+  } else {
+    worstCaseQuantileTwoSe <- NULL
+  }
+
+  bootResultPower <- computeBootObj(
+    values = stopEsWorstCase,
+    alpha = alpha,
+    nBoot = nBoot,
+    objType = "betaFromEValues"
+  )
+
+  if (estimateImpliedTarget) {
+    logImpliedTarget <- mean(log(finalEsWorstCase))
+    bootResultImpliedTarget <- computeBootObj(
+      values = finalEsWorstCase,
+      nBoot = nBoot,
+      objType = "logImpliedTarget"
+    )
+    logImpliedTargetTwoSe <- 2 * bootResultImpliedTarget[["bootSe"]]
+  } else {
+    logImpliedTarget <- logImpliedTargetTwoSe <- NULL
+  }
+
+  return(
+    list(
+      worstCasePower = currentWorstCasePower,
+      worstCasePowerTwoSe = 2 * bootResultPower[["bootSe"]],
```

Comment:
As a matter of symmetry it would be nice to define this above. The point is compare the last assignments logImpliedTarget = logImpliedTarget, logImpliedTargetTwoSe = logImpliedTargetTwoSe